lib/model/model.go

```diff
@@ -800,18 +800,18 @@ type FolderCompletion struct {
 	NeedItems     int
 	NeedDeletes   int
 	Sequence      int64
-	Accepted      bool
+	RemoteState   string
```

Comment:
I think there's some value in keeping this as a `remoteFolderState` here still. Currently it's only used in the api, where the conversion to string is needed, but say some service listens to the events and wants to act on certain states -> the typed values are nicer to use there.

Reply:
If I just change it to `remoteFolderState`, the event contains an integer though, the `.String()` conversion is not used.

Reply:
Yes, then it needs some special handling for conversion to json. There's already a (kinda weird) version of that in `.Map` on `FolderCompletion` where you could call `.String()` on it. The proper way would be to add a `.MarshalText` method that just returns `.String()` for `remoteFolderState`.

Reply:
Should we expor the type as well then? `RemoteFolderState`?  It's currently only used in the model itself.